### PR TITLE
Add monochromatic effects: Pulse, Random

### PR DESCRIPTION
**Pull request by ferbar**
# What does this implement/fix? 

Add monochromatic ligth effects pulse and random, add RGB light effect pulse.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (this will require users to update their yaml configuration files to keep working)

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs/pull/1067
  
# Test Environment

- [ ] ESP32
- [x] ESP8266
- [ ] Windows
- [ ] Mac OS
- [x] Linux

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
  - platform: monochromatic
[snip]
    effects:
      - pulse:
          transition_length: 1s
      - random:
```

# Explain your changes

Adding a pulse effect, add random effect to monochromatic light;
Adding the pulse effect to rgb light, too.

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).
esphome/esphome-docs/pull/1067

### CODE REVIEW

File: esphome/components/light/effects.py [hunk 1]
```diff
@@ -153,6 +154,52 @@ def automation_effect_to_code(config, effect_id):
...
+    {
+        cv.Optional(
+            CONF_TRANSITION_LENGTH, default="7.5s"
+        ): cv.positive_time_period_milliseconds,
+        cv.Optional(
+            CONF_UPDATE_INTERVAL, default="10s"
+        ): cv.positive_time_period_milliseconds,
+    },
+)
+@register_monochromatic_effect(
```
Comment (jesserockz) [inline esphome/components/light/effects.py:202]: `@register_monochromatic_effect` also adds to the RGB registry so you don't need both. Same for pulse

File: esphome/components/light/base_light_effects.h [hunk 2]
```diff
@@ -11,6 +11,40 @@ inline static float random_cubic_float() {
...
 }
 
+/// Pulse effect.
+class PulseLightEffect : public LightEffect {
+ public:
+  explicit PulseLightEffect(const std::string &name) : LightEffect(name) {}
+
+  void apply() override {
+    const uint32_t now = millis();
+    if (now - this->last_color_change_ < this->transition_length_) {
```
Comment (jesserockz) [inline esphome/components/light/base_light_effects.h:21]: I think this is supposed to be `update_interval_`
```suggestion
    if (now - this->last_color_change_ < this->update_interval_) {
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/light/effects.py (hunk 1 above)
Comment (ferbar) [reply esphome/components/light/effects.py]: Thanks, fixed

File: esphome/components/light/base_light_effects.h (hunk 2 above)
Comment (ferbar) [reply esphome/components/light/base_light_effects.h]: Fixed.

**Review by jesserockz — APPROVED** [review]

**Comment by jesserockz** [comment]
Looking good thanks. Just needs a docs PR please =)

**Comment by ferbar** [comment]
Added doc PR
Thanks

**Pull request merged**